src/app/events/poap/AddUserForEvent.ts

```diff
@@ -49,47 +50,52 @@ export const addUserToDb = async (
 		// event change is not related to event parameter
 		return;
 	}
-
-	// Check if user joined channel
-	if (newState.channelId === channel.id) {
-		await updateUserForPOAP(member, db, channel, true).catch(e => LogUtils.logError('failed to capture user joined for poap', e));
-	} else {
-		await updateUserForPOAP(member, db, channel, false).catch(e => LogUtils.logError('failed to capture user left for poap', e));
+	if (newState.deaf) {
+		await updateUserForPOAP(member, db, channel, false, true).catch(e => LogUtils.logError('failed to capture user joined for poap', e));
```

Comment:
'failed to capture user who attempted to join for POAP'

Reply:
ah this is also a log entry